### ISSUE TITLE
Find & remove very large files

### ISSUE THREAD

**Issue by IanMayo** [debrief/debrief#1299]
Our repo is huge.

Many files in the repo come from external projects, and can be re-created if necessary.  

So, if they're not being used in the current build, these files could be deleted from the repo history.

Here's a tool that claims to delete all kinds of things from a git repo:
https://rtyley.github.io/bfg-repo-cleaner/

It would be really useful if you could collate a list of the very largest files in the repo history, then we can discuss/verify them, then delete them.

**Comment by pecko** [comment]
Just FYI:
Debrief size before building is about 921 MB. This way we will decrease that. It will be good in order to have < !GB github limit.
It will help slightly to decrease size for travis limit, but not too much because mostly size is when creating products (Mac, Linux 64-bit, Win 32- and 64-bit).

Anyway, I will find the list later.

**Comment by IanMayo** [comment]
@pecko - I'm keen to remove large files from the **history**, as well as the current build.    This should make it *much* quicker to clone the repo to a local drive.

I hope we've already tried to remove large files from the current build.

**Comment by pecko** [comment]
Yes, I understand you. Just minor explanation related to travis.

> I hope we've already tried to remove large files from the current build.

Right. We have already tried do it from history too when we moving from bitbucket. Of course, we can remove some files yet.

BTW
You can clone with --depth=20 that will clone only 20 latest commits (travis uses 50).
Of course, we will remove big unnecessary files from all history.

**Comment by pecko** [comment]
I have made the file: bigtosmall.txt based on http://naleid.com/blog/2012/01/17/finding-and-purging-big-files-from-git-history. 
It is on Google drive: https://drive.google.com/open?id=0B6-y5Qg6UAUaRk9qZVZzOVUxZGc

Now, we can remove file using bfg (your link; it is easier) or filer-branch.
I suppose that we can remove:

- trunk, branches, deploy, org.mwc.debrief.test and satc directories
- eclipselink.jar
- a lot files from contribs
- sources from org.mwc.cmap.geotools
....

**Comment by IanMayo** [comment]
Hi there, want to be really sure we're doing this right = so will go through in detail tomorrow.

**Comment by pecko** [comment]
Of course.
We have not done anything yet. We just create the list of files.

**Comment by pecko** [comment]
BTW
Currently our clean repository (without build): 800870K (738M)
